core/src/main/java/com/graphhopper/routing/InstructionsFromEdges.java

```diff
@@ -196,19 +204,42 @@ public void next(EdgeIteratorState edge, int index, int prevEdgeId) {
                     .setDirOfRotation(deltaOut)
                     .setExited();
 
+            doublePrevName = prevName;
             prevName = name;
             prevAnnotation = annotation;
 
         } else {
             int sign = getTurn(edge, baseNode, prevNode, adjNode, annotation, name);
 
             if (sign != Instruction.IGNORE) {
```

Comment:
Not really related, but we should move all instruction related code into a new package and make this constant package protected?